registry/handlers/referrers.go

```diff
@@ -109,29 +111,14 @@ func (h *referrersHandler) generateReferrersList(ctx context.Context, subjectDig
 			if err != nil {
 				return err
 			}
-			artifactManifest, ok := man.(*ociartifact.DeserializedManifest)
-			if !ok {
-				// The PUT handler would guard against this situation. Skip this manifest.
-				return nil
-			}
-			extractedArtifactType := artifactManifest.ArtifactType
-			// filtering by artifact type or bypass if no artifact type specified
-			if artifactType == "" || extractedArtifactType == artifactType {
-				desc, err := blobStatter.Stat(ctx, referrerDigest)
-				if err != nil {
-					return err
-				}
-				desc.MediaType, _, _ = man.Payload()
-				artifactDesc := v1.Descriptor{
-					MediaType:    desc.MediaType,
-					Size:         desc.Size,
-					Digest:       desc.Digest,
-					ArtifactType: extractedArtifactType,
-					Annotations:  artifactManifest.Annotations,
-				}
-				referrers = append(referrers, artifactDesc)
+			switch man.(type) {
+			case *ocischema.DeserializedManifest:
+				return getImageReferrers(ctx, blobStatter, referrerDigest, man, &referrers)
+			case *ociartifact.DeserializedManifest:
+				return getArtifactReferrers(ctx, blobStatter, referrerDigest, man, &referrers, artifactType)
+			default:
+				return fmt.Errorf("referrers not supported for this manifest type")
```

Comment:
Should it return `nil` to keep the previous behavior?

Reply:
changed accordingly.